### ISSUE TITLE
Realm: divide-by-zero in create_weighted_subspaces when the sum of weights is 0

### ISSUE THREAD

**Issue by magnatelee** [StanfordLegion/legion#816]
Here is a reproducer:

```
import "regent"                                                                                                                                                                                                                                             
                                                                                                                                                                                                                                                            
local c = regentlib.c                                                                                                                                                                                                                                       
                                                                                                                                                                                                                                                            
task main()
  var is = ispace(int1d, 0)
  var _is = __raw(is)
  var cs = ispace(int1d, 1)
  var _cs = __raw(cs)
  var num_colors = 1
  var colors = array(([int1d](0)):to_domain_point())
  var weights = array([int32](0))
  c.legion_index_partition_create_by_weights(
      __runtime(), __context(), _is, colors,
      weights, num_colors, _cs, 1, 1)
end

regentlib.start(main)
```

**Comment by magnatelee** [comment]
Pushed the following workaround that @streichler suggested until we have a proper fix:
```
diff --git a/runtime/realm/deppart/partitions.cc b/runtime/realm/deppart/partitions.cc
index 80b3625bf..ff52f5f32 100644
--- a/runtime/realm/deppart/partitions.cc
+++ b/runtime/realm/deppart/partitions.cc
@@ -312,6 +312,7 @@ namespace Realm {
     assert(weights.size() == count);
     for(size_t i = 0; i < count; i++)
       total_weight += weights[i];
+    if(total_weight == 0) total_weight = 1;
 
     // dense case is easy(er)
     if(dense()) {
```

**Comment by streichler** [comment]
This should be fixed with commit 71da370.  @magnatelee can you take a look and confirm it works for you?

**Comment by magnatelee** [comment]
Confirmed the fix.
